### PR TITLE
Fix issue #620

### PR DESCRIPTION
**Pull request by rimadoma**
Fixes issue #620 and adds tests for `DefaultMedian`.

### CODE REVIEW

File: src/main/java/net/imagej/ops/stats/DefaultQuantile.java [hunk 1]
```diff
@@ -61,25 +62,30 @@
...
 		final ArrayList<Double> statistics = new ArrayList<>();
 
-		final Iterator<I> it = input.iterator();
-		while (it.hasNext()) {
-			statistics.add(it.next().getRealDouble());
+		input.forEach(i -> statistics.add(i.getRealDouble()));
+
+		final int k = (int) (statistics.size() * quantile);
+		double result = select(statistics, k);
+		if (statistics.size() % 2 == 0) {
```
Comment (imagejan) [inline src/main/java/net/imagej/ops/stats/DefaultQuantile.java]: How about quantiles other than the median? Is the convention to *always* take the mean of two values when the sample is even-sized?

The explanation [on wikipedia](https://en.wikipedia.org/wiki/Quantile#Even-sized_population) suggests that averaging is done for the *median* only, but not for any other quantiles (e.g. quartiles etc.).

The [R Documentation](https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/quantile) lists 9 possible algorithms (see `type` argument) to compute the quantile.

Comment (rimadoma) [reply src/main/java/net/imagej/ops/stats/DefaultQuantile.java]: Good question. According to [apache.commons version](https://commons.apache.org/proper/commons-math/javadocs/api-3.6/org/apache/commons/math3/stat/descriptive/rank/Percentile.html) of quantile: _"There are several commonly used methods for estimating percentiles (a.k.a. quantiles) based on sample data. For large samples, the different methods agree closely, but when sample sizes are small, different methods will give significantly different results."_ 

As I'm not familiar with quantiles myself, maybe the wisest thing is to leave `DefaultQuantile` as is, and add this implementation to `DefaultMedian`. What do you think @imagejan?

**Comment by rimadoma** [comment]
@imagejan So I changed the PR so that only `DefaultMedian` changes, not `DefaultQuantile`. Since there seems to be several accepted methods of calculating quantile, I reckon `DefaultQuantile` works the way it does for a reason, and should be left alone.

**Comment by ctrueden** [comment]
Thank you very much, @rimadoma!

@gselzer Please forward port this work to imagej-ops2.

**Comment by gselzer** [comment]
@ctrueden work is ported on the imagej/imagej-ops2/init branch. Thanks @rimadoma!

**Pull request merged**